### PR TITLE
support op_namescope for better debugging

### PR DESCRIPTION
**Pull request by panyx0718**
image

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/parallel_executor_test_base.py [hunk 1]
```diff
@@ -76,6 +76,7 @@ def run_executor(exe, feed, fetch_list, program=None):
                 exec_strategy.use_experimental_executor = True
 
             build_strategy = fluid.BuildStrategy()
+            build_strategy.debug_graphviz_path = "/tmp/graph_viz"
```
Comment (typhoonzero) [inline python/paddle/fluid/tests/unittests/parallel_executor_test_base.py]: use python `tempfile`?

File: python/paddle/fluid/unique_name.py [hunk 2]
```diff
@@ -77,3 +77,49 @@ def guard(new_generator=None):
...
+        return self._name
+
+
+_name_scope = NameScope()
+
+
+@contextlib.contextmanager
+def name_scope(prefix=None):
+    # TODO(panyx0718): Only [0-9a-z].
+    assert prefix, "namescope prefix cannot be empty."
```
Comment (typhoonzero) [inline python/paddle/fluid/unique_name.py]: maybe add comments in this API that namescope is only used for debugging, should not use in other case like graph optimization etc.

Comment (panyx0718) [reply python/paddle/fluid/unique_name.py]: Done

File: python/paddle/fluid/tests/unittests/parallel_executor_test_base.py (hunk 1 above)
Comment (panyx0718) [reply python/paddle/fluid/tests/unittests/parallel_executor_test_base.py]: removed. This was only used for debugging

**Review by typhoonzero — DISMISSED** [review]

**Review by qingqing01 — APPROVED** [review]

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**